src/PackageSelection/PackageSelection.php

```diff
@@ -159,17 +164,40 @@ public function select(Composer $composer, $verbose)
             $repos = $this->filterRepositories($repos);
         }
 
-        foreach ($repos as $repo) {
-            try {
-                $pool->addRepository($repo);
-            } catch (\Exception $exception) {
-                if (!$this->skipErrors) {
-                    throw $exception;
-                }
+        $packageSelection = $this;
+
+        // Event loop for parallel tasks
+        Loop::run(function () use (&$repos, $packageSelection, $pool) {
+            $workerPool = new DefaultPool();
 
-                $this->output->writeln(sprintf("<error>Skipping Exception '%s'.</error>", $exception->getMessage()));
+            $coroutines = [];
+            foreach ($repos as $repo) {
+                $coroutines[] = function () use ($workerPool, $pool, $repo) {
+                    $pool = yield $workerPool->enqueue(new AddRepoTask($pool, $repo));
```

Comment:
is it safe to modify this object across process boundaries?

Reply:
There's no synchronization between processes. Objects are serialized and unserialized in the worker, but never updated again.

Reply:
hm ok, so this seems to be no working solution then.

what could be an alternative approach (without refactoring the whole application?) ;)

Reply:
You may be able to write a process that you communicate with as progress is made. See the [process.php](https://github.com/amphp/parallel/blob/master/examples/process.php) example that communicates with a process defined by [blocking-process.php](https://github.com/amphp/parallel/blob/master/examples/blocking-process.php).